include/seqan3/alignment/matrix/detail/trace_iterator_banded.hpp

```diff
@@ -0,0 +1,99 @@
+ */
+
+#pragma once
+
+#include <seqan3/alignment/matrix/detail/trace_iterator_base.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A trace iterator for banded trace matrices.
+ * \ingroup alignment_matrix
+ *
+ * \tparam matrix_iter_t The wrapped matrix iterator; must model seqan3::detail::two_dimensional_matrix_iterator and
+ *                       the iterator must be a trace matrix, i.e.
+ *                       `std::same_as<value_type_t<matrix_iter_t>, trace_directions>` must evaluate to `true`.
+ *
+ * \details
+ *
+ * This iterator follows a given trace in a banded trace matrix.
+ */
+template <two_dimensional_matrix_iterator matrix_iter_t>
+class trace_iterator_banded : public trace_iterator_base<trace_iterator_banded<matrix_iter_t>, matrix_iter_t>
+{
+private:
+
+    static_assert(std::same_as<value_type_t<matrix_iter_t>, trace_directions>,
+                  "Value type of the underlying iterator must be trace_directions");
+
+    //!\brief The type of the base class.
+    using base_t = trace_iterator_base<trace_iterator_banded<matrix_iter_t>, matrix_iter_t>;
+
+    //!\brief Befriend base class.
+    friend base_t;
+
+public:
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr trace_iterator_banded() = default; //!< Defaulted.
+    constexpr trace_iterator_banded(trace_iterator_banded const &) = default; //!< Defaulted.
+    constexpr trace_iterator_banded(trace_iterator_banded &&) = default; //!< Defaulted.
+    constexpr trace_iterator_banded & operator=(trace_iterator_banded const &) = default; //!< Defaulted.
+    constexpr trace_iterator_banded & operator=(trace_iterator_banded &&) = default; //!< Defaulted.
+    ~trace_iterator_banded() = default; //!< Defaulted.
+
+    /*!\brief Constructs from the underlying trace matrix iterator indicating the start of the trace path.
+     * \param[in] matrix_iter The underlying matrix iterator.
+     */
+    constexpr trace_iterator_banded(matrix_iter_t const matrix_iter) noexcept : base_t{matrix_iter}
+    {}
+
+    /*!\brief Constructs from the underlying trace matrix iterator indicating the start of the trace path.
+     * \tparam other_matrix_iter_t The underlying matrix iterator type of `other`; the condition
+     *                             `std::constructible_from<matrix_iter_t, other_matrix_iter_t>` must evaluate to `true`.
+     * \param[in] other The underlying matrix iterator.
+     */
+    template <two_dimensional_matrix_iterator other_matrix_iter_t>
+    //!\cond
+        requires std::constructible_from<matrix_iter_t, other_matrix_iter_t>
+    //!\endcond
+    constexpr trace_iterator_banded(trace_iterator_banded<other_matrix_iter_t> const other) noexcept : base_t{other}
+    {}
+    //!\}
+
+private:
+
+    //!\copydoc seqan3::detail::trace_iterator_base::go_left
+    constexpr void go_left(matrix_iter_t & iter) const noexcept
+    {
+        iter -= matrix_offset{row_index_type{-1}, column_index_type{1}};
```

Comment:
Why always `-=`? 
Should be equivalent to `iter += matrix_offset{row_index_type{1}, column_index_type{-1}};`.

Reply:
It is, but is it easier to read? From the compiler point of view this shouldn't make a difference. But another coder will see that something is decremented so it is going back. That  was the intention to use subtraction.

Reply:
For me this line was confusing when doing the test by hand.
I first forgot, that those two functions are overloaded;
when I then checked here, it was kinda confusing that there is a `-1` but you actually have to go one down...so `+1`. For me everything was inverted.

I guess if you're used to `-=`, it comes more naturally. 
And I think it's mainly this line because of the double negation; the other functions are more straight forward; so we can leave it as it is,

Reply:
yes it is only the overload in the banded case that makes this more difficult to read since the actual column in the band is shifted one cell up. Maybe I leave a comment though